dbs-virtio-devices: fix balloon 32bit error

HerlinCoder opened the pull request:
We fixes a problem that offset should be parsed as u32 instead of u64 in virtio-balloon.

Fixes: #284 

We add Madvise error type to expose error when setting madvise on guest memory region.

Fixes: #285

HerlinCoder committed:
dbs-virtio-devices: add Madvise error type

We add Madvise error type to expose error when setting madvise on guest
memory region.

Fixes: #285

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

studychao commented on crates/dbs-virtio-devices/Cargo.toml:
```diff
@@ -1,29 +1,29 @@
 [package]
-name = "dbs-virtio-devices"
```
Is this chanegd by `cargo fmt`?

HerlinCoder replied:
It might be formatted by vscode... seems the new one is by alphabet. If this should be reverted pls tell me and i will fix it.

studychao replied:
This is not used to what we have done and it's not consistent with other Cargo.toml file...
Please make sure that if that is introduced by newer rust fmt tool, then it's ok to keep it.
But if this is introduced by VScode which is not a rust standard, please consider revert it.

HerlinCoder replied:
OK i will revert iit

studychao commented on crates/dbs-virtio-devices/src/lib.rs:
```diff
@@ -180,6 +180,9 @@ pub enum Error {
     InsertMmap(vm_memory::mmap::Error),
+    /// Failed to set madvise on guest memory region.
+    #[error("failed to set madvice() on guest memory region")]
+    Madvise(#[source] nix::Error),
```
Not seeing any place to use this error type, will be introduced later?

HerlinCoder replied:
Due to some historical reason, dragonball device manager need to use virtio error. So it should have a new type for Madvise.

studychao commented on crates/dbs-virtio-devices/src/balloon.rs:
```diff
@@ -57,15 +57,11 @@ const INFLATE_QUEUE_AVAIL_EVENT: u32 = 0;
 const DEFLATE_QUEUE_AVAIL_EVENT: u32 = 1;
 //// New descriptors are pending on the virtio queue.
```
extra slash here?

HerlinCoder replied:
I will fix it

studychao commented on crates/dbs-virtio-devices/src/balloon.rs:
```diff
@@ -343,12 +339,10 @@ impl<AS: DbsGuestAddressSpace, Q: QueueT + Send, R: GuestMemoryRegion>
                 );
 
-                let guest_addr = pfn << VIRTIO_BALLOON_PFN_SHIFT;
+                let guest_addr = (pfn as u64) << VIRTIO_BALLOON_PFN_SHIFT;
```
Why do you add `as u64` here? To avoid `.into()` below?

HerlinCoder replied:
gpa here should be u64 otherwise it might be overflowed

studychao replied:
make sense!

HerlinCoder committed:
dbs-virtio-devices: fix balloon pfn using u32

We fixes a problem that offset should be parsed as u32 instead of
u64 in virtio-balloon.

Fixes: #284

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>
